WalletWasabi/Crypto/Strobe.cs

```diff
@@ -46,24 +48,32 @@ private Strobe128(byte[] state, StrobeFlags flags, byte beginPosition, byte posi
 
 		public void AddAssociatedMetaData(byte[] data, bool more)
 		{
+			Guard.NotNull(nameof(data), data);
+
 			BeginOperation(StrobeFlags.M | StrobeFlags.A, more);
 			Absorb(data);
 		}
 
 		public void AddAssociatedData(byte[] data, bool more)
 		{
+			Guard.NotNull(nameof(data), data);
+
 			BeginOperation(StrobeFlags.A, more);
 			Absorb(data);
 		}
 
 		public byte[] Prf(int count, bool more)
 		{
+			Guard.InRangeAndNotNull(nameof(count), count, 0, int.MaxValue);
```

Comment:
what do you think about making this an unsigned int? does that mess with `new byte[count]`? in the C implementation it uses an `ssize_t`, which is signed, but since we're not doing any arithmetic on length values and don't need to represent -1... but i'm not sure if there are gotchas with using uints like that in C# so i'm not sure